### PR TITLE
Merge Master -> exp/update-toolset

### PR DESCRIPTION
**Pull request by cdmihai**
- converted the nuget resolver tests to the new build logic
- force xlf sync to make tests pass. Before, the original English resx file was directly used, whereas now the original resx is converted to xlf, which is converted into the temporary resx that gets used. This means we can't batch sync any longer

### CODE REVIEW

File: src/NuGetSdkResolver.UnitTests/project.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿{
```
Comment (dsplaisted) [inline src/NuGetSdkResolver.UnitTests/project.json]: Delete the project.json file

File: src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,80 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- <PropertyGroup>
```
Comment (dsplaisted) [inline src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj]: Remove comments

File: src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    <TargetFrameworks>$(RuntimeOutputTargetFrameworks)</TargetFrameworks>
+    <PlatformTarget>$(RuntimeOutputPlatformTarget)</PlatformTarget>
+    <RootNamespace>NuGet.MSBuildSdkResolver.UnitTests</RootNamespace>
+    <AssemblyName>NuGet.MSBuildSdkResolver.UnitTests</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IsPackable>false</IsPackable>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
```
Comment (dsplaisted) [inline src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj]: Remove comments (and ItemGroup which will then be empty)

File: src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  </PropertyGroup> -->
+
+  <PropertyGroup>
+    <TargetFrameworks>$(RuntimeOutputTargetFrameworks)</TargetFrameworks>
+    <PlatformTarget>$(RuntimeOutputPlatformTarget)</PlatformTarget>
+    <RootNamespace>NuGet.MSBuildSdkResolver.UnitTests</RootNamespace>
+    <AssemblyName>NuGet.MSBuildSdkResolver.UnitTests</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IsPackable>false</IsPackable>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
```
Comment (dsplaisted) [inline src/NuGetSdkResolver.UnitTests/NuGet.MSBuildSdkResolver.UnitTests.csproj]: I'd delete this line (allowing `EnableDefaultCompileItems` to be true), and then remove the `Compile` items below that are already in the project folder.  The rest of the projects don't consistently do this, but if we're adding a new project I think we may as well let it use the default globs.

File: src/Tasks/Microsoft.Common.CurrentVersion.targets [hunk 5]
```diff
@@ -1545,15 +1545,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
...
-
-      Likewise if the dependency is for build ordering instead of an assembly reference
-      (ReferenceOutputAssembly=false), skip the checks since we can't know what TF
-      the output would need to be compatible with.
    -->
    <ItemGroup>
-      <_MSBuildProjectReferenceExistent
-        Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and
-                   ('%(Extension)' == '.vcxproj' or '%(ReferenceOutputAssembly)' == 'false')">
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' == '' and '%(Extension)' == '.vcxproj'">
```
Comment (cdmihai) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets]: FYI @rainersigwald :)

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Full please
@dotnet-bot test Windows_NT Build for CoreCLR  please

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Full please

**Pull request merged**